homeassistant/components/device_tracker/icloud.py

```diff
@@ -47,24 +47,22 @@ def keep_alive(now):
         api.authenticate()
         _LOGGER.info("Authenticate against iCloud")
 
-    track_utc_time_change(hass, keep_alive, second=0)
+    track_utc_time_change(hass, keep_alive, second=0, minute=4)
 
     def update_icloud(now):
         """Authenticate against iCloud and scan for devices."""
         try:
             # The session timeouts if we are not using it so we
             # have to re-authenticate. This will send an email.
-            api.authenticate()
+            keep_alive(None)
             # Loop through every device registered with the iCloud account
             for device in api.devices:
                 status = device.status()
                 location = device.location()
                 # If the device has a location add it. If not do nothing
                 if location:
                     see(
-                        dev_id=re.sub(r"(\s|\W|')",
-                                      '',
-                                      status['name']),
+                        dev_id=slugify(status['name'].replace(' ', '', 99)),
```

Comment:
Only to have consistent names with the old implementation (the regex removed spaces and non-words)

Personal preference I would not remove the spaces, but this will change everyone's know_devices and rules based on those names